CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+# How to contribute to CSMIM
+
+![Steps for filing an issue](docs/contribute-add.svg)
+
+Would you like to see an interface standardized as a CSMIM object type?
+Or do you wish to enhance or change an existing type?
+
+- **Search for existing CSMIM object types first** before requesting to
+  create a new one.
+
+- **Search the issues for existing discussions of your request.** Contribute
+  to an existing discussion if possible, instead of opening a new issue.
+
+- **Open an issue, describe your requirements and label it an enhancement.**
+  If there are existing object types similar to what you need, mention them
+  and explain what is missing for you and how the type should be enhanced.
+  Note that we cannot change types once they are standardized, we can only
+  replace them by newer types, with backwards-compatibility if possible.
+
+Do you have a very concrete proposal for an object type?
+
+- **Usually it is better to open an issue** and describe your requirements
```

Comment:
lets insist on discussions in issues first (and discuss only implementations in pull requests). So instead of "it is better", I would demand to open an issue anyways. Pull requests shall be linked to issues.